src/server/scripts/EasternKingdoms/Karazhan/karazhan.cpp

```diff
@@ -586,6 +586,60 @@ class at_karazhan_side_entrance : public OnlyOnceAreaTriggerScript
     }
 };
 
+class at_karazhan_mirkblood_approach : public AreaTriggerScript
+{
+public:
+    at_karazhan_mirkblood_approach() : AreaTriggerScript("at_karazhan_mirkblood_approach") {}
+
+    bool OnTrigger(Player* player, AreaTrigger const* /*trigger*/) override
+    {
+        LOG_ERROR("sql.sql", "hit approach");
+        if (InstanceScript* instance = player->GetInstanceScript())
+        {
+            LOG_ERROR("sql.sql", "approach-script");
+            if (instance->GetBossState(DATA_MIRKBLOOD) != DONE)
+            {
+                LOG_ERROR("sql.sql", "approach-state");
+                if (Creature* mirkblood = instance->GetCreature(DATA_MIRKBLOOD))
+                {
+                    LOG_ERROR("sql.sql", "approach-creature");
+                    instance->SetData(DATA_MIRKBLOOD_APPROACH, player->GetGUID().GetCounter());
```

Comment:
```suggestion
                    mirkblood->AI()->Talk(SAY_APPROACH, player);
```

Might need to move the enum around because SAY_APPROACH is not accessible in the instance file, it is also fine to have his areatriggers in his script, too

Reply:
Moved the AT scripts to his file as to avoid moving around the speech enum.